x/signutil/sign.go

```diff
@@ -0,0 +1,124 @@
+package signutil
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+)
+
+type MsgArbitrarySignature struct {
+	Message []byte         `json:"message"`
+	Signer  sdk.AccAddress `json:"signer"`
+}
+
+func (m MsgArbitrarySignature) Route() string {
+	return ModuleName
+}
+
+func (m MsgArbitrarySignature) Type() string {
+	return "arbitrary_signature"
+}
+
+func (m MsgArbitrarySignature) ValidateBasic() error {
+	if len(m.Message) == 0 {
+		return fmt.Errorf("empty msg")
+	}
+	if m.Signer.Empty() {
+		return fmt.Errorf("missing signer")
+	}
+	return nil
+}
+
+// GetSignBytes implements sdk.Message
+func (m MsgArbitrarySignature) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(m))
+}
+
+// GetSigners implements sdk.Message
+func (m MsgArbitrarySignature) GetSigners() []sdk.AccAddress { return []sdk.AccAddress{m.Signer} }
+
+// getTxCmd clubs together all the CLI tx commands
+func getTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
+	configTxCmd := &cobra.Command{
+		Use:                        storeKey,
+		Short:                      fmt.Sprintf("%s transactions subcommands", storeKey),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	configTxCmd.AddCommand(flags.PostCommands(
+		sign(cdc),
+		verify(cdc),
+	)...)
+	return configTxCmd
+}
+
+func sign(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sign",
+		Short: "create a signature for a file or an arbitrary string",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txBuilder := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+			// get file
+			file, err := cmd.Flags().GetString("file")
+			if err != nil {
+				return err
+			}
+			// get text
+			text, err := cmd.Flags().GetString("text")
+			if err != nil {
+				return err
+			}
+			if text != "" && file != "" {
+				return fmt.Errorf("only text or file can be specified")
+			}
+			buf := new(bytes.Buffer)
+			switch true {
+			case text != "":
+				buf.WriteString(text)
+			case file != "":
+				f, err := os.Open(file)
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+				_, err = io.Copy(buf, f)
+				if err != nil {
+					return err
+				}
+			default:
+				return fmt.Errorf("either file or text flag must be specified")
+			}
+			msg := MsgArbitrarySignature{Message: buf.Bytes(), Signer: cliCtx.GetFromAddress()}
```

Comment:
@fdymylja, we want the message to be human readable.  What's the argument against that?